File: Lab5_13_2/Moving_Material_Point.py
from abc import ABC, abstractclassmethod
import math


class Moving_Material_Point(ABC):

    @abstractclassmethod
    def find_x(self, t: float):
        raise NotImplementedError

    @abstractclassmethod
    def find_y(self, t: float):
        raise NotImplementedError

    @staticmethod
    def find_distance(t: float, point, other):
        return math.sqrt(math.pow((point.find_x(t) - other.find_x(t)), 2) + math.pow((point.find_y(t) - other.find_y(t)), 2))

File: Lab5_13_2/Lab5_13_2/Lab5_13_2/MovingMaterialPoint.cs
using System;
namespace Lab5_13_2
{
    public abstract class MovingMaterialPoint
    {
        public abstract double findX(double t);
        public abstract double findY(double t);
        public static double findDistance(double t, MovingMaterialPoint point, MovingMaterialPoint other)
        {
            return Math.Sqrt(Math.Pow(point.findX(t) - other.findX(t), 2) + Math.Pow(point.findY(t) - other.findY(t), 2));
        }
    }
}
